omod/src/main/java/org/openmrs/module/fhir/providers/RestfulConditionResourceProvider.java

```diff
@@ -84,4 +86,18 @@ public Condition findConditionByUuid(
 	public MethodOutcome createFHIRCondition(@ResourceParam Condition condition) {
 		return MethodOutcomeBuilder.buildCreate(conditionResource.createFHIRCondition(condition));
 	}
+
+	/**
+	 * @see org.openmrs.module.fhir.resources.FHIRConditionResource#updateFHIRCondition(org.hl7.fhir.dstu3.model.Condition, java.lang.String)
+	 */
+	@Update
+	public MethodOutcome updateFHIRCondition(@ResourceParam Condition condition, @IdParam IdType uuid) {
+		try {
+			return MethodOutcomeBuilder.buildUpdate(conditionResource.updateFHIRCondition(condition, uuid.getIdPart()));
+		}
+		catch (Exception e) {
+			throw new InvalidRequestException("Following exception occured: " + e.getMessage() + " " +
```

Comment:
How do you tell that the cause of this exception is the request being invalid?

Reply:
I thought the only reason why this method `MethodOutcomeBuilder.buildUpdate` could fail is because of an invalid request

Reply:
Then why catch the generic Exception in the first place? Why not InvalidRequestException?